src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs

```diff
@@ -28,14 +28,21 @@ internal class TracerProviderBuilderSdk : TracerProviderBuilder
     {
         private readonly List<InstrumentationFactory> instrumentationFactories = new List<InstrumentationFactory>();
 
+        private readonly TracerProviderOptions options;
         private readonly List<BaseProcessor<Activity>> processors = new List<BaseProcessor<Activity>>();
         private readonly List<string> sources = new List<string>();
         private readonly Dictionary<string, bool> legacyActivityOperationNames = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
         private ResourceBuilder resourceBuilder = ResourceBuilder.CreateDefault();
         private Sampler sampler = new ParentBasedSampler(new AlwaysOnSampler());
 
-        internal TracerProviderBuilderSdk()
+        internal TracerProviderBuilderSdk(TracerProviderOptions options)
         {
+            this.options = options ?? new TracerProviderOptions();
+
+            if (options.SetErrorStatusOnException)
+            {
+                this.AddProcessor(new ExceptionProcessor());
```

Comment:
we can catch the notsupportedexception throw from Exceptionprocessor (internal) and instead throw a more user friendly one from here? A message like "SetErrorStatusOnException is not supported on this platform."

Reply:
Good point, updated.